index.d.ts

```diff
@@ -1,5 +1,13 @@
 import { Action, ActionCreator, AnyAction, StoreEnhancer, Store } from 'redux';
 
+export type PromiseResult<T> = T extends Promise<infer U> ? U : never;
```

Comment:
This looks a lot more complex than your original change. Can you explain why you needed to change so much? I'm not that familiar with typescript myself and I've mostly been relying on contributors to keep the types correct. I figured the only change you needed was the one I mentioned in the original PR.